### DISCUSSION TITLE
How and why does root directory change so often in a repo? Especially Go project.

### DISCUSSION THREAD

**Discussion by kcaashish** [LazyVim/LazyVim#1714] [Q&A]
In my case, we have a Golang repo in which every time I move into a file inside `cmd` the root directory shifts to the `cmd/dir` instead of the root directory of the project I started nvim from.  If I select a file, say in `pkg`, then the root changes to the one I opened the nvim from. 

Go's folder structure has `main.go` files inside of each subdirectory in the `cmd` directory. May be that is the cause? 

It affects neotree and specially troubling is it affects telescope and find-files. What can I do about this? 

Also, in some other cases, no matter how far inside the project I am, the cmd doesn't change from the outer root.

**Comment by dpetka2001** [comment]
Maybe this #1307 could help you, though not sure if it 100% applies to you. Not familiar with Go.

**Reply by kcaashish** [reply]
Thanks! This thread was actually useful for me.

**Comment by folke** [comment]
Set `vim.g.root_spec = {"cwd"}` in your `options.lua`

**Reply by kcaashish** [reply]
Thank you!

**Reply by kostas-petrakis** [reply]
Thanks! I was trying to figure out what was happening :)
